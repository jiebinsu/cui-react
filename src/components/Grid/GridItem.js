import React from "react";
import PropTypes from "prop-types";
import classNames from "../../utils/classNames";

const GridItem = ({
  className,
  size,
  offset,
  sm,
  smOffset,
  md,
  mdOffset,
  lg,
  lgOffset,
  xlg,
  xlgOffset,
  collapse,
  children,
  ...props
}) => {
  const classes = classNames("coop-l-grid__item", [
    size && `coop-l-grid__item--${size}`,
    offset && `coop-l-grid__item--offset-${offset}`,
    sm && `coop-l-grid__item--small-${sm}`,
    smOffset && `coop-l-grid__item--small-${smOffset}`,
    md && `coop-l-grid__item--medium-${md}`,
    mdOffset && `coop-l-grid__item--medium-offset-${mdOffset}`,
    lg && `coop-l-grid__item--large-${lg}`,
    lgOffset && `coop-l-grid__item--large-offset-${lgOffset}`,
    xlg && `coop-l-grid__item--xlarge-${xlg}`,
    xlgOffset && `coop-l-grid__item--xlarge-offset-${xlgOffset}`,
    collapse && "coop-l-grid__item--collapse",
    className,
  ]);

  return (
    <div className={classes} {...props}>
      {children}
    </div>
  );
};

GridItem.defaultProps = {
  className: "",
  size: null,
  offset: null,
  sm: null,
  smOffset: null,
  md: null,
  mdOffset: null,
  lg: null,
  lgOffset: null,
  xlg: null,
  xlgOffset: null,
  collapse: false,
  children: null,
};

GridItem.propTypes = {
  className: PropTypes.string,
  size: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  offset: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  sm: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  smOffset: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  md: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  mdOffset: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  lg: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  lgOffset: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  xlg: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  xlgOffset: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  collapse: PropTypes.bool,
  children: PropTypes.node,
};

export default GridItem;
