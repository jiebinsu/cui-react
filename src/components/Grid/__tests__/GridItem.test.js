import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import GridItem from "../GridItem";

describe("GridItem", () => {
  it("should render correctly", () => {
    const wrapper = mount(<GridItem />);
    expect(() => wrapper.unmount()).not.toThrow();

    const tree = renderer.create(<GridItem />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should allow size and offset attributes", () => {
    const SizesAndOffsets = () => (
      <>
        <GridItem size={6} />
        <GridItem offset={2} />
        <GridItem sm={6} />
        <GridItem smOffset={2} />
        <GridItem md={6} />
        <GridItem mdOffset={2} />
        <GridItem lg={6} />
        <GridItem lgOffset={2} />
        <GridItem xlg={6} />
        <GridItem xlgOffset={2} />
        <GridItem
          size={6}
          offset={4}
          sm={5}
          smOffset={3}
          md={4}
          mdOffset={2}
          lg={3}
          lgOffset={1}
          xlg={2}
          xlgOffset={1}
        />
      </>
    );
    const tree = renderer.create(<SizesAndOffsets />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should allow collapse attribute", () => {
    const Collapse = () => (
      <>
        <GridItem collapse />
      </>
    );
    const tree = renderer.create(<Collapse />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should allow children", () => {
    const tree = renderer.create(<GridItem>test</GridItem>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
