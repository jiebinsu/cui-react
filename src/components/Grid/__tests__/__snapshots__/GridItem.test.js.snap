// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridItem should allow children 1`] = `
<div
  className="coop-l-grid__item"
>
  test
</div>
`;

exports[`GridItem should allow collapse attribute 1`] = `
<div
  className="coop-l-grid__item coop-l-grid__item--collapse"
/>
`;

exports[`GridItem should allow size and offset attributes 1`] = `
Array [
  <div
    className="coop-l-grid__item coop-l-grid__item--6"
  />,
  <div
    className="coop-l-grid__item coop-l-grid__item--offset-2"
  />,
  <div
    className="coop-l-grid__item coop-l-grid__item--small-6"
  />,
  <div
    className="coop-l-grid__item coop-l-grid__item--small-2"
  />,
  <div
    className="coop-l-grid__item coop-l-grid__item--medium-6"
  />,
  <div
    className="coop-l-grid__item coop-l-grid__item--medium-offset-2"
  />,
  <div
    className="coop-l-grid__item coop-l-grid__item--large-6"
  />,
  <div
    className="coop-l-grid__item coop-l-grid__item--large-offset-2"
  />,
  <div
    className="coop-l-grid__item coop-l-grid__item--xlarge-6"
  />,
  <div
    className="coop-l-grid__item coop-l-grid__item--xlarge-offset-2"
  />,
  <div
    className="coop-l-grid__item coop-l-grid__item--6 coop-l-grid__item--offset-4 coop-l-grid__item--small-5 coop-l-grid__item--small-3 coop-l-grid__item--medium-4 coop-l-grid__item--medium-offset-2 coop-l-grid__item--large-3 coop-l-grid__item--large-offset-1 coop-l-grid__item--xlarge-2 coop-l-grid__item--xlarge-offset-1"
  />,
]
`;

exports[`GridItem should render correctly 1`] = `
<div
  className="coop-l-grid__item"
/>
`;
